### DISCUSSION TITLE
warning for non string query keys

### DISCUSSION THREAD

**Discussion by shrynx** [TanStack/query#501] [General]
Hi ! 
Thanks for the library, was longing for a solution like this for a while.

I got bit by this, had a query key like `['user', userId, 'posts']` but where `userId` was a number.
When using `useQuery` it worked and the userId was shows as a string in devtools but when trying to refetch using `queryCache.refetchQueries` it didn't work and silently failed.

although this is clearly mentioned in the documentation, would also be nice if the library warns on non string query key variables. In my case i had forgotten userId is not string.

Another solution would be to have automatically convert to string using `JSON.stringify`(that's what i end up doing most cases)
Also i think `queryCache.refetchQueries` when provided with query keys that don't exists should also warn

Happy to discuss further and make a PR if needed

**Comment by hetpatel33** [comment]
@shrynx You can use an array with a string and any number of serializable objects.

**Comment by shrynx** [comment]
yup i totally can and should but as i mentioned, this more for adding warning when they keys in array is not string as they silently fail during `queryCache.refetchQueries` but doesn't fail for `useQuery`

**Comment by ee0pdt** [comment]
Isn't this what Typescript is for?

**Comment by shrynx** [comment]
that seems fair, if `queryCache.refetchQueries` wasn't silently failing and would throw runtime error that would make sense
but runtime error maybe aggressive so a warning during development seems good ergonomics

Also useQuery automatically coerces `userId` to string.
i wish the behaviour for keys for both `queryCache.refetchQueries`  and `useQuery` was similar
